node.go

```diff
@@ -164,6 +172,21 @@ func (node *Node) Visit(opts *Options) (dirs, files int) {
 	return
 }
 
+func (node *Node) match(pattern string, opt *Options) bool {
+	var prefix string
+	if opt.IgnoreCase {
+		prefix = "(?i)"
+	}
+
+	search := node.Name()
+	re, err := regexp.Compile(prefix + pattern)
+	if strings.Contains(re.String(), "*") {
+		search = node.path
+	}
+
+	return err == nil && re.FindString(search) != ""
+}
```

Comment:
Actually I think there's a very good reason not to use `re.MatchString`, because it will return `true` even if it matched a `(blank)` string `""`, which while is correct in regex since a `*` wild card will match the preceding token 0 or more times. However, at the very least `tree`, and pretty much all Linux commands w/ regex, treat the blank string as a non-matching result. So if we're building a go implementation of `tree` I think it should match that implementation.

Reply:
I think the other suggested changes are fine and I've pushed those

Reply:
Thanks for the explanation.